.pre-commit-config.yaml

```diff
@@ -23,40 +23,27 @@ repos:
           - '-DSPGLIB_WITH_TESTS=ON'
         stages: [ manual ]
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.3.0
+    rev: v4.5.0
     hooks:
       - id: trailing-whitespace
       - id: end-of-file-fixer
       - id: check-yaml
       - id: check-added-large-files
-  - repo: https://github.com/pycqa/flake8
-    rev: 5.0.4
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.12
     hooks:
-      - id: flake8
-        files: python\/.*
-        args:
-          - "--max-line-length=88"
-          - "--ignore=E203,W503"
-  - repo: https://github.com/psf/black
-    rev: 22.6.0
-    hooks:
-      - id: black
-        files: python\/.*
-        args: [ "--line-length=88" ]
-  - repo: https://github.com/pycqa/pydocstyle
-    rev: 6.1.1
-    hooks:
-      - id: pydocstyle
-        files: python/spglib
-        additional_dependencies: [ "toml" ]
-  - repo: https://github.com/pycqa/isort
-    rev: 5.12.0
+      - id: ruff
+        args: [ --fix, --exit-non-zero-on-fix ]
+        types_or: [python, pyi]
+      - id: ruff-format
+        types_or: [python, pyi]
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.8.0
     hooks:
-      - id: isort
-        files: python\/.*
-        args: [ "--profile", "black" ]
+    - id: mypy
+      exclude: "database"
```

Comment:
`database/*.py` is not actively maintained now

Reply:
Better to move the exclude to `pyproject.toml`

Reply:
Without it, `pre-commit run --all` complains about `dataset/ptg2symbol.py`. git-commit and `pre-commit run --all` seem to use different settings.

Reply:
Even when you set [`exclude`](https://mypy.readthedocs.io/en/stable/config_file.html#confval-exclude)? I'll have to try that locally

Reply:
Yes, `pre-commit run --all` ignores it...